pennylane/math/multi_dispatch.py

```diff
@@ -896,3 +896,39 @@ def detach(tensor, like=None):
         return np.to_numpy(tensor)
 
     return tensor
+
+
+@multi_dispatch()
+def set_index(array, idx, val, like=None):
+    """Set the value at a specified index in an array.
+    Calls ``array[idx]=val`` and returns the updated array unless JAX.
+
+    Args:
+        array (tensor_like): array to be modified
+        idx (int, tuple): index to modify
+        val (int, float): value to set
+
+    Returns:
+        a new copy of the array with the specified index updated to ``val``.
+
+    Whether the original array is modified is interface-dependent.
+
+    .. note:: TensorFlow EagerTensor does not support item assignment
+    """
+    # if passed an index of the form (DynamicJaxprTracer, DynamicJaxprTracer),
+    # multidispatch only sees the Tuple and doesn't recognize the interface as jax
+    try:

```

Comment:
I think here you do not need that or you should not use multi_dispatch. Because the decorator is doing the job for you. `@multi_dispatch()` is the same as `@multi_dispatch([0, 1, 2])` and it sets `like` to the right interface. I think here you need to specify that `id` can be a tensor_list `@multi_dispatch([0, 1, 2], tensors_list=[1])``

Reply:
Thanks! I hadn't understood how that worked, that is much nicer :)